gap/oper.gi

```diff
@@ -334,7 +396,23 @@ end);
 
 InstallMethod(DigraphReverseEdges, "for an immutable digraph and a list",
 [IsImmutableDigraph, IsList],
-{D, E} -> MakeImmutable(DigraphReverseEdges(DigraphMutableCopy(D), E)));
+function(D, E)
+  local changed, DMutable, edge;
+  changed := false;
+  DMutable := DigraphMutableCopy(D);
+  for edge in E do
+    DigraphReverseEdge(DMutable, edge);
+    if edge[1] <> edge[2] then  # The edge could be reversed
+      changed := true;
+    fi;
+  od;
+
+  if changed = true then
+    return MakeImmutable(DMutable);
```

Comment:
Again, could we figure out whether the condition is true *without* creating a copy first?